src/YesSql.Core/Provider/IDbProviderOptions.cs

```diff
@@ -0,0 +1,11 @@
+﻿using YesSql.Core.Services;
+
+namespace YesSql.Core.Provider
+{
+    public interface IDbProviderOptions
+    {
+        string ProviderName { get; set; }
```

Comment:
Thanks for detailed explanation, if that's the case I think Orchard CMS is already have a way to list those providers from DI or some where else, which is out of the scope here

But still I see there's a benefit to have a `IDbProviderOptions`, perhaps the 'ProviderName` is not that useful here, but at least the use can know the current provider without digging into the `DbConnectionFactory`, also `IDbProviderOptions` may useful to keep track of table prefix

> Also where do you specify that a provider accepts connection strings or not?

this is done in the extension methods `UseSqlServer("connectionString")`